packages/core/src/scheduler/task.rs

```diff
@@ -33,7 +33,11 @@ impl Scheduler {
     ///
     /// Spawning a future onto the root scope will cause it to be dropped when the root component is dropped - which
     /// will only occur when the VirtuaalDom itself has been dropped.
-    pub fn spawn(&self, scope: ScopeId, task: impl Future<Output = ()> + 'static) -> TaskId {
+    pub fn spawn(
+        &self,
+        scope: ScopeId,
+        task: impl Future<Output = ()> + 'static,
+    ) -> Option<TaskId> {
```

Comment:
Would prefer to keep returning `TaskId` so we can move it to a `JoinHandle` -esque API in the future.

This would involve creating the waker as you've already done on line 57, but not returning None. 

If you wanted to take this is a step further, adding some sort of JoinHandle methods onto TaskId would be useful by checking with the current Dioxus runtime.